### PR TITLE
Update documents in "tutorials" for ORT format models, mobile and web

### PR DESCRIPTION
**Pull request by fs-eire**
**Description**: Update documents in "tutorials" for ORT format models, mobile and web. This change includes:
-	Add a new entry for “Deploy ORT format model”. This will include full step of preparing ORT format models.
-	Move common contents from “Deploy ORT mobile” to “Deploy ORT format model”, if it is not specified to mobile.
-	Both “Deploy on web” and “deploy on mobile” point to “Deploy ORT format model”.
-	In “Deploy ORT format model”, if a step is different from mobile and web, make them separated and create links for them.

**Motivation and Context**
- Why is this change required? What problem does it solve?
    ORT format model was only used in ORT mobile. So the old documents do not separate documents about "ORT format model" and "ORT mobile". Now ORT Web also uses the model format so we need to make the documents clear.

Review link:
-	Official: [Tutorials](https://onnxruntime.ai/docs/tutorials/)
-	My change: [Tutorials (fs-eire.github.io)](https://fs-eire.github.io/onnxruntime/docs/tutorials/)

### CODE REVIEW

File: docs/tutorials/ort-format-model/model-execution.md [hunk 1]
```diff
@@ -49,11 +50,16 @@ OrtEnvironment env = OrtEnvironment.getEnvironment();
...
+```js
+// TBD
+```
+
 ### Load ORT format model from an in-memory byte array
 
 If a session is created using an input byte array containing the ORT format model data, by default we will copy the model bytes at the time of session creation to ensure the model bytes buffer is valid.
 
-You may also enable the option to use the model bytes directly by setting the Session Options config `session.use_ort_model_bytes_directly` to `1`, this may reduce the peak memory usage of ONNX Runtime Mobile, you will need to guarantee that the model bytes are valid throughout the lifespan of the ORT session using the model bytes.
+You may also enable the option to use the model bytes directly by setting the Session Options config `session.use_ort_model_bytes_directly` to `1`, this may reduce the peak memory usage of ONNX Runtime Mobile, you will need to guarantee that the model bytes are valid throughout the lifespan of the ORT session using the model bytes. For ONNX Runtime Web, this option is set by default.
```
[the remark this answers is not in the document]
Comment (fs-eire) [reply docs/tutorials/ort-format-model/model-execution.md:62]: yes I think so. The option is introduced later than first version of ORT mobile. In my opinion, if it is not for backward compatibility, ORT mobile should set this flag by default.

File: docs/build/web.md [hunk 2]
```diff
@@ -75,7 +75,7 @@ To get all build artifacts of ONNX Runtime WebAssembly, it needs 4 times of buil
 
 ### Minimal Build Support
 
-ONNX Runtime WebAssembly can be built with flag `--minimal_build`. This will generate smaller artifacts and also have a less runtime memory usage. An ORT format model is required. A detailed instruction will come soon. See also [ONNX Model Conversion](../tutorials/mobile/model-conversion.html).
+ONNX Runtime WebAssembly can be built with flag `--minimal_build`. This will generate smaller artifacts and also have a less runtime memory usage. An ORT format model is required. A detailed instruction will come soon. See also [ONNX Model Conversion](../tutorials/ort-format-model/model-conversion.html).
```
[the remark this answers is not in the document]
Comment (fs-eire) [reply docs/build/web.md:78]: thanks for catching the link error. will fix

File: docs/tutorials/ort-format-model/model-execution.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (gwang-msft) [reply docs/tutorials/ort-format-model/model-execution.md:64]: For Mobile we have future plan about this (to further reduce memory usage), so this is optional for now.

**Comment by gwang-msft** [comment]
Can we change the "Deploy ORT format model for mobile device and web" to "Deploy on mobile device and web", the term "ORT format" can be confusing, we can add a separate section to introduce the "ORT format model"

File: docs/tutorials/ort-format-model/model-execution.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (fs-eire) [reply docs/tutorials/ort-format-model/model-execution.md:64]: @gwang-msft is this plan also benefiting web as well? do you mind sharing more information about this?

**Comment by fs-eire** [comment]
> Can we change the "Deploy ORT format model for mobile device and web" to "Deploy on mobile device and web", the term "ORT format" can be confusing, we can add a separate section to introduce the "ORT format model"

When I try to revise the contents, I feel like we should introduce a term for this concept. The current "Deploy ORT format model for mobile device and web" section includes information about the "ORT format". Users who follow this tutorials are aware of this ( they generate the .ort file ). Maybe "ORT format model" is not the best term, but we need a name for it instead of removing it from docs.

File: docs/tutorials/ort-format-model/overview.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+parent: Deploy ORT format model for mobile device and web
+grand_parent: Tutorials
+has_children: false
+nav_order: 1
+---
+{::options toc_levels="2" /}
+
+# Overview
+
+The execution environment on Mobile devices, and for web browsers, have fixed memory and disk storage. Therefore, it is essential that any AI execution library is optimized to consume minimun resources in terms of disk footprint and memory. ONNX Runtime was enhanced to target these size contrained environments. These enhancments are packaged in the ONNX Runtime Mobile offering.
```
Comment (skottmckay) [inline docs/tutorials/ort-format-model/overview.md:12]: >contrained [](http://example.com/codeflow?start=288&length=10)

constrained

File: docs/tutorials/ort-format-model/overview.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+The pre-built package includes support for selected operators and ONNX opset versions based on the requirements for popular models. If you choose to use the pre-built package you will not need a development environment to perform a custom build of ONNX Runtime, however the binary size will be larger than if you do a custom build with just the operators required by your model/s. Your model can only use the opsets and operators supported by the pre-built package.
+
+### Available pre-built packages
+
+| Platform | Package location | Included Execution Providers |
+|----------|------------------|----------|
+| Android | onnxruntime-mobile package in Maven  | CPU Execution Provider <br>NNAPI Execution Provider |
+| iOS (preview) | onnxruntime-mobile CocoaPod (C/C++ APIs) <br>onnxruntime-mobile-objc CocoaPod (Objective-C API) | CPU Execution Provider <br>CoreML Execution Provider |
+| Web (preview) | onnxruntime-web package in NPM | WebAssembly Execution Provider <br>WebGL Execution Provider |
```
Comment (skottmckay) [inline docs/tutorials/ort-format-model/overview.md:37]: I believe iOS can be considered released and no longer preview

**Comment by skottmckay** [comment]
> | 1.7 | 2021-03-03 | rel-1.7.2 |

Can we update with 1.9 info while we're at it?

---
Refers to: docs/tutorials/ort-format-model/initial-setup.md:91 in 5882507. [](commit_id = 5882507d908c22fec37cc00f7a5773a783207759, deletion_comment = False)

File: docs/tutorials/ort-format-model/packaging-and-deploy.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
+title: Packaging and deploy
+parent: Deploy ORT format model for mobile device and web
```
Comment (skottmckay) [inline docs/tutorials/ort-format-model/packaging-and-deploy.md:3]: Is this page going to have content before checkin? If not should we skip adding it until we have the content?

**Review by natke — CHANGES_REQUESTED** [review]
We don't want to change paths in the docs, as it interferes with our search  indexing and performance. If we absolutely need to chnage path, then we should add a redirect from the original path

**Comment by fs-eire** [comment]
> We don't want to change paths in the docs, as it interferes with our search indexing and performance. If we absolutely need to chnage path, then we should add a redirect from the original path

I understand the concerns. however one purpose of this change is to separate the concept of "ORT format model" and "ORT mobile" so IMO the change of the doc structure and paths is necessary.

File: docs/tutorials/ort-format-model/packaging-and-deploy.md (hunk 5 above)
Comment (fs-eire) [reply docs/tutorials/ort-format-model/packaging-and-deploy.md]: yes i will update content before checkin

**Comment by fs-eire** [comment]
> > | 1.7 | 2021-03-03 | rel-1.7.2 |
> 
> Can we update with 1.9 info while we're at it?
> 
> Refers to: docs/tutorials/ort-format-model/initial-setup.md:91 in [5882507](https://github.com/microsoft/onnxruntime/commit/5882507d908c22fec37cc00f7a5773a783207759). [](commit_id = [5882507](https://github.com/microsoft/onnxruntime/commit/5882507d908c22fec37cc00f7a5773a783207759), deletion_comment = False)

added

File: docs/tutorials/ort-format-model/overview.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+parent: Deploy on mobile device and web
+grand_parent: Tutorials
+has_children: false
+nav_order: 1
+---
+{::options toc_levels="2" /}
+
+# Overview
+
+The execution environment on Mobile devices, and for web browsers, have fixed memory and disk storage. The execution scenario on web browsers have strict memory consumption and network bandwidths requirement. Therefore, it is essential that any AI execution library is optimized to consume minimun resources in terms of disk footprint, memory and network usage (both model size and binary size). ONNX Runtime was enhanced to target these size constrained environments. These enhancments are packaged in the ONNX Runtime Mobile and web offering.
```
Comment (gwang-msft) [inline docs/tutorials/ort-format-model/overview.md:12]: ... library is optimized to consume **minimum** resources in terms of disk ...

...constrained environments. These **enhancements** are ...

File: docs/build/reduced.md [hunk 7]
```diff
@@ -11,7 +11,7 @@ For applications where package binary size is important, ONNX Runtime provides o
 
 To reduce the compiled binary size of ONNX Runtime, the operator kernels included in the build can be reduced to just the kernels required by your model/s.
 
-For deployment on mobile devices specifically, please read more detailed guidance on [Deploy ONNX Runtime Mobile](../tutorials/mobile/).
+For deployment on mobile devices and web specifically, please read more detailed guidance on [Deploy on mobile device and web](../tutorials/ort-format-model/).
```
Comment (gwang-msft) [inline docs/build/reduced.md:14]: Do we still want to use the folder `ort-format-model`? Or something like `mobile-web`?

**Comment by gwang-msft** [comment]
LGTM for mobile part

File: docs/build/reduced.md (hunk 7 above)
Comment (fs-eire) [reply docs/build/reduced.md:14]: as mentioned by @natke we try to minimize the change of paths so let's keep it "ort-format-model" in URL.

**Review by manashgoswami — APPROVED** [review]

**Comment by natke** [comment]
We're going to add a redirect for the changes paths #9750.

I am also going to spend a little more time reviewing these changes for SEO best practice. Thank you for keeping the docs up to date!

**Comment by natke** [comment]
A couple of observations:
- The title of the section in the `ort-format-model` pages is "Deploy on Mobile and Web". Whilst titles and article URLs do not need to match exactly, it will help navigability and discoverability if they are mostly aligned. This is about the terminology we use in the product. Unless there is a good reason to do otherwise, we should use settle on one term for one concept.
- A lot of the information in this section is background context rather than a tutorial. Ideally a tutorial is a set of steps along with instructional material that will demonstrate and teach a user how to do the thing that is represented by the title of the page (and this would be aligned with what someone would search when they are trying to solve a problem).
- Some of the pages contain a lot of information with a lot of different customer scenarios and some of them just a brief snippet (e.g. limitations)
- Some of the information is reference material e.g. specifications of options for conversion scripts
- Do we need https://fs-eire.github.io/onnxruntime/docs/tutorials/ort-format-model/model-execution.html? It seems that the ONNX Runtime inference APIs can be used without modification with ORT models?

Recommendations:
- [ ] Change URL to deploy-mobile-web
- [ ] Refactor the articles to separate context information and instructional material
- [ ] Refactor the articles to align each page with a customer scenario
- [ ] Update diagrams to include web as well as mobile
- [ ] Move custom-build.md into the reference section
- [ ] Move build instructions into the build section
- [ ] Move EP instructions into a customer scenario for Android / iOS / web

**Comment by skottmckay** [comment]
> * Do we need https://fs-eire.github.io/onnxruntime/docs/tutorials/ort-format-model/model-execution.html? It seems that the ONNX Runtime inference APIs can be used without modification with ORT models?

It documents additional steps that are ORT format model specific. I think adding that to the general API documentation would be noisy so a separate page here makes sense to me.

**Comment by natke** [comment]
Superceded by #9854

**Pull request closed without merging**